File: androidApp/src/main/kotlin/dev/lildua/oddly/ui/navigation/Destination.kt
package dev.lildua.oddly.ui.navigation

/**
 * Every screen the app can show.
 *
 * Only [TabDestination]s appear in the bottom bar (spec §5: exactly four
 * destinations); everything else is a full-screen child route.
 */
sealed interface Destination {

    // --- Onboarding ---
    data object Splash : Destination
    data object Onboarding : Destination
    data object ChooseInterest : Destination
    data object NotificationPermission : Destination

    // --- Child routes ---
    data class ChallengeDetail(val challengeId: String) : Destination
    data class ChallengeComplete(val challengeId: String) : Destination
    data class ShareCard(val challengeId: String) : Destination
    data object AnotherChallenge : Destination
    data object ChooseCategory : Destination
    data object AllChallenges : Destination
    data object Calendar : Destination
    data object Streak : Destination
    data object Quotes : Destination
}

/** The four bottom-navigation destinations. */
enum class TabDestination(val label: String) : Destination {
    HOME("Hôm nay"),
    JOURNEY("Hành trình"),
    STATISTICS("Thống kê"),
    SETTINGS("Cài đặt"),
}

File: iosApp/iosApp/Navigation/Destination.swift
import Foundation

/// The four bottom-navigation destinations (spec §5: exactly four, everything
/// deeper is a child route).
enum TabDestination: String, CaseIterable, Hashable {
    case home
    case journey
    case statistics
    case settings

    var label: String {
        switch self {
        case .home: return "Hôm nay"
        case .journey: return "Hành trình"
        case .statistics: return "Thống kê"
        case .settings: return "Cài đặt"
        }
    }

    var icon: OddlyIcon {
        switch self {
        case .home: return .sparkle
        case .journey: return .journey
        case .statistics: return .stats
        case .settings: return .settings
        }
    }
}

/// Child routes pushed onto a tab's navigation stack. Screens that are modal by
/// nature — the reward celebration and the share card — are presented as covers
/// and sheets instead, so they are not listed here.
enum Destination: Hashable {
    case challengeDetail(challengeId: String)
    case anotherChallenge
    case chooseCategory
    case allChallenges
    case calendar
    case streak
    case quotes
}

/// The linear onboarding sequence (spec §2.1).
enum OnboardingStep: Hashable {
    case chooseInterest
    case notificationPermission
}
